### ISSUE TITLE
Need more properties

### ISSUE THREAD

**Issue by xaque208** [xaque208/puppet-jail#12]
For a postgres jail, the following is required.

```
iocage set allow_sysvipc=1 db1
```

Exposing an interface to this seems prudent.

**Comment by igalic** [comment]
we can do that with a `jail_property` and have them prefetched per jail, and for the default one, too…

**Comment by RainbowHackerHorse** [comment]
Was this ever worked on?
Enabling the use of resource limits on a jail via puppet would be very nice to have as well.
Ultimately, this module should be exposing most, if not all, of the options iocage allows one to set per-jail.
I'm not particularly well-versed in ruby, but I'd be happy to help where I can.

**Comment by xaque208** [comment]
I don't think this got captured yet.  The upstream `iocage` is in a bit of flux at the moment.  We've got a pending PR https://github.com/xaque208/puppet-jail/pull/32 that could use some testing (and probably some code changes) before we merge.  This is to track the upstream changes in the `iocage` project itself, but I've yet to get a completely working provider to handle the new iocage code.

**Comment by xaque208** [comment]
To point specifically at the PR I mentioned:

https://github.com/xaque208/puppet-jail/blob/4901f413d582ed0a1f2b0d353972548d502c8957/lib/puppet/type/jail.rb#L119

That line allows for any properties to be set for the jail, so that if iocage adds a property, we don't need to also make a change in the module to support it.  So that's the future, but the today story is less good.

**Comment by RainbowHackerHorse** [comment]
@xaque208 Wow, fast response is fast.
Yeah, I noticed the py-iocage stuff is still...very much in flux. Even though it's deprecated, this is why I strictly stick to legacy `iocage` even if it's not exactly supported anymore.
On that note, I opened #34 with some added parameters that should work with the existing 3.x branch codebase for the iocage legacy stuff. Understandable if you don't want to merge it, being that you plan to track py-iocage exclusively, but I just figured I'd push my localized changes for my infra upstream to you in case you wanted to include them in a final 3.x puppet-jail release or something :)
Thanks for working on such a great & useful module btw!

**Comment by RainbowHackerHorse** [comment]
Oh nice. Yeah, that sounds like it'd be much easier than the kind of a hassle way you have to do things currently.
(On the bright side, while it's not much, needing to add those params caused me to write my first lines of ruby today, so hello 4th language!)

**Comment by xaque208** [comment]
Congrats on the new language!  That's always fun (for me).

I don't see any reason why we can't include the iocage_legacy provider forever.  Its been deleted in the PR I mentioned (but could easily be added back in before merge), and someone forked iocage into a new project `iocell` which is based on the shell version before the python work, so I'd considered renaming/copying the iocage_legacy provider to a new provider called iocell, which could still be useful for folks who don't want to make the transition.

I've got about a dozen jails personally that were under management of this module, but due to the changes in iocage, the process has completely broken for me.  Its been a slightly frustrating series of events, so I'd still like to get a working story here.

**Comment by RainbowHackerHorse** [comment]
Thanks! I'm on my 4th language in 8 months after starting a new job, not including learning puppet's DSL.
Bit crazy, but I'm enjoying it :)

I hadn't heard of iocell, I'll have to check it out. While iocage legacy was nice, there were always a few things I wanted to see fixed or added.

That being said, for those of use who still rely on the old shell version, I'm sure more folks beyond just myself would appreciate the continued inclusion of iocage-legacy support in the module, at least until py-iocage matures, or iocell becomes a solid competitor.

Once I've got more ruby under my belt, I'd also be happy to take time to contribute on a semi-regular basis, as I'm puppetizing all my personal infra and I see myself getting a lot of use from this module in the near future. The least I can do is contribute back!
